libs/utils/utils.js

```diff
@@ -882,7 +882,7 @@ export const getMepEnablement = (mdKey, paramKey = false) => {
   return getMdValue(mdKey);
 };
 
-let imsLoaded;
+let imsLoaded = window.adobeIMS?.initialized;
```

Comment:
I'm not fully convinced this will work every time, but we can adapt if needed. Here's an use-case: user has `adobeid` defined and is loading imslib script (window.adobeIMS?.initialized would be false) and at the same time we're executing `loadIms` method and we override the `adobeid` object and load imslib.min.js from www.adobe.com. The result is that ims would get configured incorrectly or might not work properly since we're loading two instances of imslib.min.js

Reply:
If we expose a key in the config that can be set by the consuming client to indicate whether they want to use their own IMS or Gnav's IMS, we can modify the code as follows:

 ```
const { useClientIms } = getConfig();
    if (useClientIms) return;

    return loadIms()
      .then(() => this.imsReady())
      .catch((e) => {
        if (e?.message === 'IMS timeout') {
          window.addEventListener('onImsLibInstance', () => this.imsReady());
          return;
        }
        lanaLog({ message: 'GNAV: Error with IMS', e, tags: 'errorType=info,module=gnav' });
      });
```
      
What are your thoughts on this approach? @narcis-radu @mokimo @salonijain3

Reply:
Let's take a step back first:

What is the exact use-case for this? Where/how did the requirement come from, and which issue is this supposed to solve?

Reply:
With standalone navigation, consumers like Adobe Home want to load their own IMS. They use IMS before rendering global navigation, as it's apparently needed for other checks as well.

Reply:
[Wiki link ](https://wiki.corp.adobe.com/pages/viewpage.action?pageId=3285416619#MilobasedGNav&FooterIntegrationinAdobeHome-QueriesforMiloteam)to their requirement request.

Reply:
So, the case is:
- The page (non-milo Adobe Home) already loads IMS
- The page embeds the standalone GNAV
- GNAV should not initiate loading IMS

I don't think we should make it a general milo pattern that consumers can define their own IMS loading mechanism. This should be steered by milo which should provide the core logic for consuming projects.

The GNAV works as standalone version and hence should add the logic separately IMO. I'd suggest adapting the [GNAV logic to check whether IMS](https://github.com/adobecom/milo/blob/stage/libs/blocks/global-navigation/global-navigation.js#L325) has already loaded

Reply:
e: I saw this is basically the same thing you proposed, i somehow was still in the mindset of the milo utils 😆

Reply:
For the adobe scenario you can be sure IMS already finished loading, hence for now I'd just move the check to the GNAV
`ims = async () => window.adobeIMS?.initialized ? this.imsReady() : loadIms()`

Reply:
Yes, This sounds good. I will update the code as so.